lld/MachO/Driver.cpp

```diff
@@ -1295,6 +1394,27 @@ static void createFiles(const InputArgList &args) {
       break;
     }
   }
+
+  if (config->readThreads) {
+    multiThreadedPageIn(deferredFiles);
+
+    DeferredFiles archiveContents;
+    std::vector<ArchiveFile *> archives;
+    for (auto &file : deferredFiles)
+      if (ArchiveFile *archive = dyn_cast<ArchiveFile>(
+              processFile(file.buffer, &archiveContents, file.path,
+                          LoadType::CommandLine, file.isLazy)))
+        archives.push_back(archive);
+
+    if (!archiveContents.empty()) {
+      multiThreadedPageIn(archiveContents);
+      for (auto *archive : archives)
+        archive->addLazySymbols();
+    }
+
+    // flush threads
+    multiThreadedPageIn(DeferredFiles());
+  }
```

Comment:
Do we need to do this? Wouldn't the threads be paging in parallel as the rest of the linker advances? I also proposed having a explicit `join` in some other comment.

Reply:
I was worried about memory being referenced going out of scope somewhere. Seemed safer to reap the threads.